spec.md

```diff
@@ -236,14 +246,20 @@ within the same JSON object.
 ### id
 
 - Type: `String`
-- Description: ID of the event. The semantics of this string are explicitly
-  undefined to ease the implementation of producers. Enables deduplication.
+- Description: Optionally identifies an event for de-duplication.
+  Absent if de-duplication is not required.
+  If present, the `source` and `id` attributes together uniquely identify the event.
 - Examples:
+  - An event counter maintained by the producer
   - A database commit ID
+  - A UUID
 - Constraints:
-  - REQUIRED
-  - MUST be a non-empty string
-  - MUST be unique within the scope of the producer
+  - OPTIONAL
+    - Producers MAY omit `id` if de-duplication is not required.
```

Comment:
The [Design Goal section](https://github.com/cloudevents/spec/blob/master/primer.md#design-goals) states:

> CloudEvents are typically used in a distributed system to allow for services to be loosely coupled during development, deployed independently, and later can be connected to create new applications.

In a loosely coupled system, a producer can't tell if de-duplication is not required (otherwise they're strongly coupled), and even if it could - can the producer tell that if it is later connected to create a new application, de-duplication won't be required in the future?